Engine/Helper.cs

```diff
@@ -656,22 +656,46 @@ public IScriptExtent GetScriptExtentForFunctionName(FunctionDefinitionAst functi
             if (null == functionDefinitionAst)
             {
                 return null;
+            }            
+            var funcNameToken = Tokens.Where(
+                token => ContainsExtent(functionDefinitionAst.Extent, token.Extent) 
+                && token.Text.Equals(functionDefinitionAst.Name));
+            if (funcNameToken.Any())
```

Comment:
You can simplify this a little bit by using `funcNameToken.FirstOrDefault()` which will return `null` if the enumerable has no items.  Your logic is perfectly fine though, no need to change it.  Just pointing out that this saves you from enumerating over the first item twice :)